### PR TITLE
top: feature ZFS ARC Header (7305)

### PR DESCRIPTION
**Pull request by fronald**
Implementation of ZFS ARC Header using kstats.
https://www.illumos.org/issues/7305

New fields are: ARC Total, MRU, MFU, Anon, Header and Other.

### CODE REVIEW

File: components/sysutils/top/Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		top
 COMPONENT_VERSION=	3.8beta1
-COMPONENT_REVISION=	1
+COMPONENT_REVISION=	2
 COMPONENT_PROJECT_URL=	http://www.unixtop.org/
 COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
 COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
```
Comment (pyhalov) [inline components/sysutils/top/Makefile:30]: Please, also update Makefile a bit: remove depend.mk (include and file), BUILD_PKG_DEPENDENCIES and add REQUIRED_PACKAGES

File: components/sysutils/top/patches/05.zfs_arc_stats.patch [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
++    
+     /* call resize to do the dirty work */
+     top_lines = display_resize();
+ 
+@@ -799,6 +816,8 @@
+ 	cpustate_columns = (int *)calloc(num_cpustates, sizeof(int));
+ 	memory_names = statics->memory_names;
+ 	num_memory = string_count(memory_names);
++        
++        arc_names = statics->arc_names;
```
Comment (pyhalov) [inline components/sysutils/top/patches/05.zfs_arc_stats.patch:217]: formatting: tabs vs spaces?

File: components/sysutils/top/patches/05.zfs_arc_stats.patch [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+ 	cpustate_total_length = 0;
+@@ -868,6 +887,19 @@
+ 	    swap_cidx[i++] = color_tag(scratchbuf);
+ 	}
+     }
++
++    /* color tags for swap */
++    if (num_arc > 0)
++    {
++	arc_cidx = (int *)malloc(num_arc * sizeof(int));
```
Comment (pyhalov) [inline components/sysutils/top/patches/05.zfs_arc_stats.patch]: Where is it free'd?

File: components/sysutils/top/patches/05.zfs_arc_stats.patch [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
++	i = 0;
++	p = strcpyend(scratchbuf, "arc.");
++	while (i < num_swap)
```
Comment (pyhalov) [inline components/sysutils/top/patches/05.zfs_arc_stats.patch]: num_arc ?

**Comment by pyhalov** [comment]
Overall, looks and feels very good

File: components/sysutils/top/Makefile (hunk 1 above)
Comment (xen0l) [reply components/sysutils/top/Makefile:30]: REQUIRED_PACKAGES are added by running "gmake REQUIRED_PACKAGES" and you should edit the Makefile afterwards and add any other deps by hand.

**Comment by fronald** [comment]
Thanks @pyhalov, i will update this things and commit again.

File: components/sysutils/top/patches/05.zfs_arc_stats.patch [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
++        arc_names = statics->arc_names;
+ 
+ 	/* calculate starting columns where needed */
+ 	cpustate_total_length = 0;
+@@ -868,6 +887,19 @@
+ 	    swap_cidx[i++] = color_tag(scratchbuf);
+ 	}
+     }
++
++    /* color tags for swap */
```
Comment (pyhalov) [inline components/sysutils/top/patches/05.zfs_arc_stats.patch]: fix comment

**Comment by fronald** [comment]
Commit updated. All done now.

**Pull request merged**